flowmachine/flowmachine/features/subscriber/label_event_score.py

```diff
@@ -0,0 +1,192 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# # License, v. 2.0. If a copy of the MPL was not distributed with this
+# # file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from shapely.geometry import shape
+from shapely.geometry.base import BaseGeometry
+from typing import List, Dict, Union, Any
+
+from .scores import EventScore
+from .subscriber_location_cluster import _JoinedHartiganCluster
+from ...core import Query
+
+
+class LabelEventScore(Query):
+    """
+    Represents a label event score class.
+
+    This class will label a table containing scores based on a labelling
+    dictionary. It allows one to specify labels which every subscriber must have.
+    This class is used to label locations based on scoring signatures in the
+    absence of other automated labelling mechanisms.
+
+    Parameters
+    ----------
+    scores : flowmachine.Query
+        A flowmachine.Query object.
+        This represents a table that contains scores which are used to label
+        a given location. This table must have a subscriber column (called subscriber).
+    labels : dict
+        A dictionary whose keys are the label names and the values geojson shapes,
+        specified hour of day, and day of week score, with hour of day score on the x axis
+        and day of week score on the y axis, where all values are in the range [-1, 1]
+    required :
+        Specifies a label which every subscriber must possess independently of
```

Comment:
From reading this it's not clear to me whether `required` must be supplied for `LabelEventScore` to function correctly. Also, what is the difference in logic depending on whether or not it is supplied?